napalm_panos/panos.py

```diff
@@ -509,28 +524,26 @@ def get_interfaces(self):
                 interface_info_xml = xmltodict.parse(self.device.xml_root())
                 interface_info_json = json.dumps(interface_info_xml['response']['result']['hw'])
                 interface_info = json.loads(interface_info_json)
-            except AttributeError:
-                interface_info = {}
+            except KeyError:
+                # loopback sub-ifs don't return a 'hw' key
```

Comment:
Can you instead check if not not get `['response']['result'].get('hw')`? I would rather not presume keyerror only means `hw` is not there.

Reply:
I see the comment states only one case, but what other response would we give if any of the other keys are missing?

What about returning a blank dictionary if `.get('hw') is None` and letting the `KeyError` exception shoot back up for any other missing key? (if that's what you meant)

Reply:
Exactly. If result wan't there, that would be a keyerror as well. The keyerror is only a subset of potential issues, so the check IMHO should only be for the specific condition you have identified. As the API changes, you could run into other issues, that this would mask.

Reply:
what about my comment here? I think it applies to this one as well. I'll make any necessary amends when we agree on something there https://github.com/napalm-automation-community/napalm-panos/pull/54#discussion-diff-162425118R629

Reply:
Agreed

Reply:
Actually did leave the `try/except` there, however only under the very specific `'loopback.' in interface and 'hw' in str(err)` does it assume it's a loopback sub-interface, for all other cases it re-raises the `KeyError` exception.